spec/models/supplier_location_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SupplierLocation do
+  it { is_expected.to belong_to(:supplier) }
+  it { is_expected.to belong_to(:location) }
+
+  it 'prevents effective_from > effective_to' do
+    expect(build(:supplier_location, effective_from: '2020-03-04', effective_to: '2020-03-03')).not_to be_valid
+  end
+
+  it 'allows effective_from == effective_to' do
```

Comment:
`allows effective_from to be greater than effective_to`
and 
`allows effective_from to be equal to effective_to`
are a bit clearer, 
not very opinionated, up to you